tests/func/parsing/test_interpolated_entry.py

```diff
@@ -259,3 +259,59 @@ def test_vars_load_partial(tmp_dir, dvc, local, vars_):
         d["vars"] = vars_
     resolver = DataResolver(dvc, tmp_dir.fs_path, d)
     resolver.resolve()
+
+
+@pytest.mark.parametrize(
+    "bool_config", [None, "store_true", "boolean_optional"]
```

Comment:
1. Maybe we only need to test two different conditions? The `None` can be tested through some other method (Test if the default value is set corretly)?
2. Because these two configurations is independent, maybe we can test them in an (I'm not sure how to describe it) instead of using a product.  ["a1b1", "a2b2"] instead of ["a1", "a2"] x ["b1", "b2"]. 
The two suggestion above can reduce the tests amount from 9 to 2.

Reply:
I updated it to 3 test cases (`["a1b1", "a2b2"]`). I still included the default in the same test, not sure if it's worth it to separate it as the body of the test would be the same